parboiled-core/src/test/scala/org/parboiled2/ActionSpec.scala

```diff
@@ -147,10 +146,10 @@ object ActionSpec extends TestParserSpec {
 
       case class Foo(i: Int, s: String)
 
-      /*FIXME: type error "`~>` producing case class (simple notation)" - new TestParser1[Foo] {
-        def targetRule = rule(push(1 :: "X" :: HNil) ~> Foo)
+      "`~>` producing case class (simple notation)" - new TestParser1[Foo] {
+        def targetRule = rule(push(1 :: "X" :: HNil) ~> Foo.apply)
```

Comment:
```suggestion
        def targetRule = rule(push(1 :: "X" :: HNil) ~> Foo.apply _)
```

`Foo.apply` would be the proper way to write it in Scala 3, but since Scala 2 doesn't eta-expand by default the common implementation for both will be `Foo.apply _`.
Sorry for the confusion :)

Reply:
Yeah, no worries, now fixed.